johnsonandjohnson/Bodiless-JS
# [DO NOT MERGE] bug(richtext, core): Prevent unnecessary re-activation of context for slate editor.

## Changes
- Memoize props provided to slate editor by RichText.
- Prevent reactivation of a context when it is already innermost.
- Modify RichText so that it does not refresh context on every change.
- Update LocalContextMenu so it only re-renders when it is actually on-screen.


## Test Instructions
- Visit a page with many slate editors
- Click on one slate editor many times and/or type into one
- Observe whether warnings appear in console about re-resolving slate editor props, and whetehr performance of slate editor is acceptable.

## Related Issues
Fixes #379
